packages/tools/client-debugger/client-debugger-chrome-extension/src/test/Utilities.ts

```diff
@@ -0,0 +1,91 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { SinonSandbox } from "sinon";
+
+import { Globals } from "../utilities";
+
+/**
+ * Wait for a listener for the specified event to be registered.
+ *
+ * @remarks
+ *
+ * Note: This must be called *before* loading the background script
+ * because listeners are registered during initialization.
+ *
+ * Note: The returned `Promise` must NOT be `await`ed until *after*
+ * loading the background script otherwise it won't resolve.
```

Comment:
Hmm why is this?

Reply:
This was adapted from the reference project I was looking at. The flow is a bit unintuitive, but it works well. The idea is we first bind a stubbed event to our sandbox to monitor the `addListener` hook. The promise then resolves when `addListener` is called, returning the callback they registered. So you need to _create_ the promise before the background script launches, such that the promise resolves when the script calls `addListener`. Then you can await the promise _after_ launching the script, and the promise will resolve assuming the script called `addListener` as expected.

Counterintuitive, but it nicely proxies the registered callback over to our tests so we can directly invoke it.

I'll take a stab at improving the docs here to make things more clear. This really shouldn't be specific to the background script either; it should be documented in terms of running before or after some computation that we expect to call `addListener`..

Reply:
Docs updated.